src/sentry/models/exporteddata.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import absolute_import
+
+from datetime import timedelta
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+from uuid import uuid4
+
+from sentry.db.models import FlexibleForeignKey, Model, sane_repr
+
+DEFAULT_EXPIRATION = timedelta(days=7)
+
+
+def generate_data_id():
+    return uuid4().hex
+
+
+class ExportedData(Model):
+    """
+    Stores references to asynchronous data export jobs being stored
+    in the Google Cloud Platform temporary storage solution.
+    """
+
+    __core__ = False
+
+    organization = FlexibleForeignKey("sentry.Organization")
+    user = FlexibleForeignKey(settings.AUTH_USER_MODEL)
+    data_id = models.CharField(max_length=32, unique=True, default=generate_data_id)
+    created_at = models.DateTimeField(default=timezone.now)
+    finished_at = models.DateTimeField(null=True)
+    expired_at = models.DateTimeField(null=True)
+    storage_url = models.URLField(null=True)
+    query_type = models.CharField(max_length=32)  # e.g.: Discover_V1, Billing, IssuesByTag
+    query_info = models.TextField()  # i.e.: sort rules, order-by's, custom settings
```

Comment:
What will actually be in here? Are you storing a json blob?

Reply:
I'm thinking a JSON blob will be generic enough, that was the plan, is there a better way of doing things like this? I'm still _very_ new to python and django, let alone the codebase

Reply:
If you're using JSON then there's an actual JSON field you can use that'll handle the conversion for you. It's probably also worth defining a schema via `jsonschema` so that we can easily validate that format.